commit 0d54ee1c7850a954026deec4cd4885f331da35cc
Author: Vegard Nossum <vegard.nossum@gmail.com>
Date:   Sat Jan 17 17:45:45 2009 +0100

    security: introduce missing kfree
    
    Plug this leak.
    
    Acked-by: David Howells <dhowells@redhat.com>
    Cc: James Morris <jmorris@namei.org>
    Cc: <stable@kernel.org>
    Signed-off-by: Vegard Nossum <vegard.nossum@gmail.com>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Adjusted to apply to Debian's 2.6.24 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.24.orig/security/keys/keyctl.c linux-source-2.6.24/security/keys/keyctl.c
--- linux-source-2.6.24.orig/security/keys/keyctl.c	2008-01-24 15:58:37.000000000 -0700
+++ linux-source-2.6.24/security/keys/keyctl.c	2009-02-09 22:43:52.000000000 -0700
@@ -253,6 +253,7 @@ long keyctl_join_session_keyring(const c
 
 	/* join the session */
 	ret = join_session_keyring(name);
+	kfree(name);
 
  error:
 	return ret;
